backend/Recodo.BLL/Services/UserService.cs

```diff
@@ -76,5 +85,37 @@ public async Task<UserDTO> GetUserById(int userId)
                 return _mapper.Map<UserDTO>(userEntity);
             }
         }
+
+        public async Task ResetPassword(string email)
+        {
+            var existUser = _context.Users.FirstOrDefault(u => u.Email == email);
+            if (existUser == null)
+            {
+                throw new ExistUserException(email);
+            }
+
+            var token = await _authService.GenerateAccessToken(existUser.Id, existUser.Email, existUser.Email);
+            string accessToken = token.AccessToken;
+            string clientHost = _configuration["ClientHost"];
+            string url = $"{clientHost}/reset-done?token={accessToken}";
+
+            await EmailService.SendEmailAsync(email, "New Password", url, _configuration);
+        }
+
+        public async Task ResetPasswordDone(UpdateUserDTO user)
+        {
+            var existUser = _context.Users.FirstOrDefault(u => u.Email == user.Email);
+            if (existUser == null)
+            {
+                throw new ExistUserException(user.Email);
+            }
+
+            var salt = SecurityHelper.GetRandomBytes();
+            existUser.Salt = Convert.ToBase64String(salt);
+            existUser.Password = SecurityHelper.HashPassword(user.PasswordNew, salt);
+
+            _context.Users.Update(existUser);
+            await _context.SaveChangesAsync();
```

Comment:
Sign in user to system after this process

Reply:
Done